### PR TITLE
[Merged by Bors] - feat(ImportGraph): add option to exclude imports from other packages

### PR DESCRIPTION
**Pull request by joneugster**
Add an option `lake exe graph --to MyProject.XY --noDep` which removes all nodes that do not start in `MyProject`.

---

Note that currently omitting `--to` is equivalent to specifying `--to Mathlib`, so there is no special case needed to handle this.

Happy to get a suggestion for a better name, maybe `--exclude-dependencies`?

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Lean/Data/NameMap.lean [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+-/
+
+namespace Lean.NameMap
+
+/--
+Filter a `NameMap`. Takes a second optional argument `modify : α → α` which allows
+to modify/filter the value of each (key, value)-pair simultaneously.
+-/
+def filter (m : NameMap α) (f : Name → Bool) (modify : α → α := id) : NameMap α :=
+  m.fold process {}
```
Comment (semorrison) [inline Mathlib/Lean/Data/NameMap.lean:21]: For consistency with other APIs, could we have `filter (f : Name → α → Bool)`, and then `filterMap (f : Name → α → Option α)`?

File: Mathlib/Lean/Name.lean [hunk 2]
```diff
@@ -48,3 +48,10 @@ def allNamesByModule (p : Name → Bool) : CoreM (Std.HashMap Name (Array Name))
       | none => return names.insert m #[n]
     else
       return names
+
+/-- Returns the very first part of a name: for `Mathlib.Data.Set.Basic` it returns `Mathlib`. -/
+def getModule (name : Name) (s := ""): Name :=
```
Comment (semorrison) [inline Mathlib/Lean/Name.lean:53]:
```suggestion
def getModule (name : Name) (s := "") : Name :=
```

**Comment by semorrison** [comment]
Could I suggest we make this useful behaviour the default, and instead have a flag `--include-deps`?

File: Mathlib/Lean/Data/NameMap.lean (hunk 1 above)
Comment (joneugster) [reply Mathlib/Lean/Data/NameMap.lean]: Done that. I only need `NameMap.filterMap` and not `NameMap.filter`, but in praxis I would more likely use `filter`
with a function `(f : Name → Bool)`:

```lean
graph := graph.filter (isPrefixOf p)
-- vs
graph := graph.filter (fun n _ => isPrefixOf p n)
```
I named the two versions `filterName` and `filter`, but maybe there is a better name.

File: Mathlib/Lean/Data/NameMap.lean [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+/--
+`filterName f m` returns the `NameMap` consisting of all
+"`key`/`val`"-pairs in `m` where `f key` returns `true`.
+
+Shorthand for `NameMap.filter (fun n _ => f n) m`.
+-/
+def filterName (f : Name → Bool) (m : NameMap α) : NameMap α :=
+  filter (fun n _ => f n) m
+
```
Comment (semorrison) [inline Mathlib/Lean/Data/NameMap.lean:34]:
```suggestion
```
You're not using it, let's just leave this out for now.

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**